clickhouse_driver/block.py

```diff
@@ -32,47 +30,130 @@ def read(self, buf):
                 self.bucket_num = read_binary_int32(buf)
 
 
-class Block(object):
+class BaseBlock(object):
+    def __init__(self, columns_with_types=None, data=None,
+                 info=None, types_check=False):
+        self.columns_with_types = columns_with_types or []
+        self.types_check = types_check
+        self.info = info or BlockInfo()
+        self.data = self.normalize(data or [])
+
+        super(BaseBlock, self).__init__()
+
+    def normalize(self, data):
+        return data
+
+    @property
+    def num_columns(self):
+        raise NotImplementedError
+
+    @property
+    def num_rows(self):
+        raise NotImplementedError
+
+    def get_columns(self):
+        raise NotImplementedError
+
+    def get_rows(self):
+        raise NotImplementedError
+
+    def get_column_by_index(self, index):
+        raise NotImplementedError
+
+    def transposed(self):
+        return list(map(tuple, zip(*self.data)))
```

Comment:
Very good! This also faster than chaining into one big tuple and slicing.

In [the very beginning](https://github.com/mymarilyn/clickhouse-driver/blob/5d5f889c2fad2ead0d7edcf47754400a2d2b747b/src/block.py#L45) it was there and later was re-writed due to performance issues. It seems that issues were in another place.